packages/ui-heading/src/Heading/styles.js

```diff
@@ -0,0 +1,126 @@
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (
+  theme,
+  themeOverride,
+  { level, color, border, margin }
+) => {
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+
+  const levelStyles = {
+    h1: {
+      fontFamily: componentTheme.h1FontFamily,
+      fontSize: componentTheme.h1FontSize,
+      fontWeight: componentTheme.h1FontWeight
+    },
+    h2: {
+      fontFamily: componentTheme.h2FontFamily,
+      fontSize: componentTheme.h2FontSize,
+      fontWeight: componentTheme.h2FontWeight
+    },
+    h3: {
+      fontFamily: componentTheme.h3FontFamily,
+      fontSize: componentTheme.h3FontSize,
+      fontWeight: componentTheme.h3FontWeight
+    },
+    h4: {
+      fontFamily: componentTheme.h4FontFamily,
+      fontSize: componentTheme.h4FontSize,
+      fontWeight: componentTheme.h4FontWeight
+    },
+    h5: {
+      fontFamily: componentTheme.h5FontFamily,
+      fontSize: componentTheme.h5FontSize,
+      fontWeight: componentTheme.h5FontWeight
+    },
+    reset: {
+      margin: '0',
+      fontSize: 'inherit',
+      fontWeight: 'inherit',
+      lineHeight: 'inherit'
+    }
+  }
+
+  const colorStyles = {
+    inherit: { color: 'inherit' },
+    primary: { color: componentTheme.primaryColor },
+    secondary: { color: componentTheme.secondaryColor },
+    'primary-inverse': { color: componentTheme.primaryInverseColor },
+    'secondary-inverse': { color: componentTheme.secondaryInverseColor }
+  }
+
+  const borderStyles = {
+    top: {
+      paddingTop: componentTheme.borderPadding,
+      borderTop: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    },
+    bottom: {
+      paddingBottom: componentTheme.borderPadding,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    },
+    none: {}
+  }
+
+  return {
+    heading: {
+      label: 'heading',
+      lineHeight: componentTheme.lineHeight,
+      '& input': {
+        outline: '0',
+        appearance: 'none',
+        boxSizing: 'border-box',
+        background: 'none',
+        border: 'none',
+        borderRadius: '0',
+        padding: '0',
+        margin: '-0.375rem 0 0 0',
+        color: 'inherit',
+        height: 'auto',
+        width: '100%',
+        lineHeight: 'inherit',
+        textAlign: 'start',
+        boxShadow: 'none',
+        display: 'block',
+        '&:focus': { outline: 'none' }
+      },
+      ...levelStyles[level],
+      ...colorStyles[color],
+      ...borderStyles[border]
+    },
+    forwardedStyleProps: {
```

Comment:
I don't think we need to do this since the View component will override the margin with the inline applied styles.

Reply:
Removed